File: backend/schemas/report.py
"""Interview report schemas."""

from typing import List

from pydantic import BaseModel, Field


class DimensionScore(BaseModel):
    dimension: str = Field(..., description="Dimension name")
    score: float = Field(..., ge=0, le=100, description="Dimension score")
    weight: float = Field(default=0.0, description="Dimension weight")
    comment: str = Field(default="", description="Dimension review")


class QuestionReview(BaseModel):
    question: str = Field(..., description="Question content")
    answer_summary: str = Field(..., description="Answer summary")
    score: float = Field(..., ge=0, le=100, description="Question score")
    comment: str = Field(default="", description="Question review")
    suggestions: str = Field(default="", description="Improvement suggestions")
    technical_accuracy: float = Field(default=0.0, ge=0, le=100, description="Technical accuracy")
    knowledge_depth: float = Field(default=0.0, ge=0, le=100, description="Knowledge depth")
    logic_expression: float = Field(default=0.0, ge=0, le=100, description="Logic expression")
    position_match: float = Field(default=0.0, ge=0, le=100, description="Position match")
    evidence: List[str] = Field(default_factory=list, description="Supporting evidence")
    issues: List[str] = Field(default_factory=list, description="Observed issues")


class LearningResource(BaseModel):
    title: str = Field(..., description="Resource title")
    type: str = Field(default="article", description="Resource type")
    description: str = Field(default="", description="Resource description")
    url: str = Field(default="", description="Resource URL")


class TrainingDay(BaseModel):
    day: int = Field(..., description="Training day")
    title: str = Field(..., description="Training topic")
    tasks: List[str] = Field(default_factory=list, description="Training tasks")


class ExpressionSummary(BaseModel):
    speech_rate: float = Field(default=0.0, description="Average speech rate")
    pause_ratio: float = Field(default=0.0, description="Average pause ratio")
    fluency_score: float = Field(default=0.0, description="Fluency score")
    confidence: float = Field(default=0.0, description="Confidence score")
    sample_count: int = Field(default=0, description="Sample count")
    clarity_score: float = Field(default=0.0, description="Speech clarity score")
    emotion_stability: float = Field(default=0.0, description="Emotional stability score")
    confidence_label: str = Field(default="", description="Confidence label")
    emotion_label: str = Field(default="", description="Emotion label")
    analysis_source: str = Field(default="", description="Analysis source")
    comment: str = Field(default="", description="Expression analysis summary")


class InterviewReport(BaseModel):
    session_id: int = Field(..., description="Interview session ID")
    position_type: str = Field(..., description="Position type")
    total_score: float = Field(..., ge=0, le=100, description="Total score")
    level: str = Field(default="", description="Evaluation level")
    dimensions: List[DimensionScore] = Field(default_factory=list, description="Dimension scores")
    question_reviews: List[QuestionReview] = Field(default_factory=list, description="Per-question reviews")
    strengths: List[str] = Field(default_factory=list, description="Strength summary")
    weaknesses: List[str] = Field(default_factory=list, description="Weakness summary")
    overall_comment: str = Field(default="", description="Overall comment")
    suggestions: List[str] = Field(default_factory=list, description="Learning suggestions")
    resources: List[LearningResource] = Field(default_factory=list, description="Recommended resources")
    training_plan: List[TrainingDay] = Field(default_factory=list, description="Training plan")
    expression_summary: ExpressionSummary | None = Field(default=None, description="Expression analysis summary")

File: frontend/src/api/report.ts
import client from './client';

export interface DimensionScore {
  dimension: string;
  score: number;
  weight: number;
  comment: string;
}

export interface QuestionReview {
  question: string;
  answer_summary: string;
  score: number;
  comment: string;
  suggestions: string;
}

export interface LearningResource {
  title: string;
  type: string;
  description: string;
  url: string;
}

export interface TrainingDay {
  day: number;
  title: string;
  tasks: string[];
}

export interface ExpressionSummary {
  speech_rate: number;
  pause_ratio: number;
  fluency_score: number;
  confidence: number;
  sample_count: number;
}

export interface InterviewReport {
  session_id: number;
  position_type: string;
  total_score: number;
  level: string;
  dimensions: DimensionScore[];
  question_reviews: QuestionReview[];
  strengths: string[];
  weaknesses: string[];
  overall_comment: string;
  suggestions: string[];
  resources: LearningResource[];
  training_plan: TrainingDay[];
  expression_summary?: ExpressionSummary | null;
}

export interface GrowthOverview {
  total_interviews: number;
  avg_score: number;
  highest_score: number;
  streak_days: number;
}

export interface TrendItem {
  date: string;
  score: number;
  position_type: string;
}

export function getReport(sessionId: number): Promise<InterviewReport> {
  return client.get(`/interviews/${sessionId}/report`);
}

export function getGrowthOverview(): Promise<GrowthOverview> {
  return client.get('/growth/overview');
}

export function getGrowthTrend(): Promise<TrendItem[]> {
  return client.get('/growth/trend');
}

export function getRecommendations(): Promise<any> {
  return client.get('/growth/recommendations');
}
